src/app/rmmapi/messagecache.ts

```diff
@@ -17,18 +17,16 @@
 // along with Runbox 7. If not, see <https://www.gnu.org/licenses/>.
 // ---------- END RUNBOX LICENSE ----------
 
-import { Injectable } from '@angular/core';
 import { Dexie } from 'dexie';
 import { MessageContents } from './rbwebmail';
 
-@Injectable()
 export class MessageCache {
     db: Dexie;
     message_version = 5;
 
-    constructor() {
+    constructor(userId: number) {
         try {
-            this.db = new Dexie('messageCache');
```

Comment:
Any thoughts on some sort of: if original "messageCache" indexdb exists, then delete (or rename!?) that before creating the new one using the userid? Otherwise we leave GBs of data lying around.

Generally we try and upgrade or at least throw out old storage we're not longer using. I suggest adding a comment which says "remove this chunk of code in 2024-06" or similar to it

Reply:
Good suggestion.

I'll add some code to delete the original `messageCache`. I'd rather not rename it because we're not 100% that its the correct message cache for the current user.